### DISCUSSION TITLE
Yield break with return value

### DISCUSSION THREAD

**Discussion by jagorath** [dotnet/csharplang#5033] [Language Ideas] [answered]
When breaking an iterator, two statements inside a code block are required, e.g.

```
if (someCondition)
{
    yield return -1;
    yield break;
}
```
With a combined return/break statement for example called "yieldlast" (or something less awkward), these four lines could be reduced to one - 

```
if (someCondition)
   yieldlast -1;
```

Or to avoid new reserved words, yield break could have an optional parameter:
 
```
if (someCondition)
   yield break -1;
```

**Comment by svick** [comment, marked as answer]
Why would it be worthwhile to add special syntax for this particular pattern? It isn't the goal of the language to make every program as short as possible.

**Reply by jagorath** [reply]
Understood, but it does make it more succinct.  What is the purpose of other new language features e.g. top-level statements and target-typed expressions other than to make for less code?

**Reply by CyrusNajmabadi** [reply]
We like to focus those efforts on changes that will benefit a very large audience. This would likely not meet that goal.

**Reply by jagorath** [reply]
Got it.  Thankyou @CyrusNajmabadi and @svick for your consideration.

**Comment by FelipeCostaGualberto** [comment]
Actually, I think that is a pretty good suggestion. Something like `yield break <value>;`
